site/zenodo_rdm/legacy/serializers/schemas.py

```diff
@@ -105,6 +105,46 @@ class ThesisSchema(Schema):
     university = SanitizedUnicode()
 
 
+class DateSchema(Schema):
+    """Date schema."""
+
+    start = EDTFDateString()
+    end = EDTFDateString()
+    type = SanitizedUnicode(
+        validate=validate.OneOf(["Collected", "Valid", "Withdrawn"])
+    )
+    description = SanitizedUnicode()
+
+    @post_dump(pass_original=True)
+    def dump_date(self, result, original, **kwargs):
+        """Dumps date field."""
+        date = original.get("date")
+        if date:
+            interval = date.split("/")
+            is_interval = len(interval) == 2
+            start = None
+            end = None
+            # EDTF level 0 specifies intervals using "/" (e.g. 2004-02-01/2005-02)
+            if is_interval:
+                start = interval[0]
+                end = interval[1]
+                result["start"] = start
+                result["end"] = end
+            else:
+                # TODO what to do here? We do not know whether the date is start, end or exact date (start == end)
```

Comment:
This case is particularly troublesome. 

In RDM, there is no concept of `start` or `end` when a date is a single date (e.g. `2023-02-02`). How do we convert it back to legacy?

[EDTF](https://www.loc.gov/standards/datetime/) implements this in level 1, however RDM implements EDTF level 0 for now.

Reply:
See my comment on the deserializer, TLDR: either make sure we have always both start/end in legacy or check how easy it is to use EDTF Level 1 in RDM.

Reply:
moved it [here](https://github.com/zenodo/zenodo-rdm/pull/347)